src/test/kotlin/org/dif/DIDCommTest.kt

```diff
@@ -40,139 +43,140 @@ class DIDCommTest {
                 .build()
         )
 
-        /*val unpack = didComm.unpack(
-            UnpackParams.Builder(packResult.packedMessage).build()
-        )*/
+        val unpack = didComm.unpack(
+            UnpackParams.Builder(packResult.packedMessage)
+                .secretResolver(BobSecretResolverMock())
+                .build()
+        )
+
+        with(unpack.metadata) {
+            assertTrue { encrypted }
+            assertTrue { authenticated }
+            assertTrue { nonRepudiation }
+            assertFalse { anonymousSender }
+            assertFalse { reWrappedInForward }
+        }
     }
 
     @Test()
-    fun `Test signed unencrypted message`() {
-        val didComm = DIDComm(DIDDocResolverMock(), SecretResolverMock())
+    fun `Test non repudiable encryption message for anonymous sender`() {
+        val didComm = DIDComm(DIDDocResolverMock(), AliceSecretResolverMock())
 
-        val msg = Message.builder("12345", mapOf("foo" to "bar"), "my-protocol/1.0")
-            .from(JWM.ALICE_DID)
-            .to(listOf(JWM.BOB_DID))
-            .build()
+        val packResult = didComm.packEncrypted(
+            PackEncryptedParams.builder(JWM.PLAINTEXT_MESSAGE, JWM.BOB_DID)
+                .protectSenderId(true)
+                .signFrom(JWM.ALICE_DID)
+                .from(JWM.ALICE_DID)
+                .build()
+        )
 
-        val packResult = didComm.packSigned(
-            PackSignedParams.builder(msg, JWM.ALICE_DID)
+        val unpack = didComm.unpack(
+            UnpackParams.Builder(packResult.packedMessage)
+                .secretResolver(BobSecretResolverMock())
                 .build()
         )
 
-        /*val unpack = didComm.unpack(
-            UnpackParams.Builder(packResult.packedMessage).build()
-        )*/
+        with(unpack.metadata) {
+            assertTrue { encrypted }
+            assertTrue { authenticated }
+            assertTrue { nonRepudiation }
+            assertTrue { anonymousSender }
+            assertFalse { reWrappedInForward }
+        }
     }
 
     @Test()
-    fun `Test plaintext message`() {
-        val didComm = DIDComm(DIDDocResolverMock(), SecretResolverMock())
-
-        val msg = Message.builder("12345", mapOf("foo" to "bar"), "my-protocol/1.0")
-            .from(JWM.ALICE_DID)
-            .to(listOf(JWM.BOB_DID))
-            .build()
+    fun `Test signed unencrypted message`() {
+        val didComm = DIDComm(DIDDocResolverMock(), AliceSecretResolverMock())
 
-        val packResult = didComm.packPlaintext(
-            PackPlaintextParams.builder(msg)
+        val packResult = didComm.packSigned(
+            PackSignedParams.builder(JWM.PLAINTEXT_MESSAGE, JWM.ALICE_DID)
                 .build()
         )
 
-        /*val unpack = didComm.unpack(
+        val unpack = didComm.unpack(
             UnpackParams.Builder(packResult.packedMessage).build()
-        )*/
+        )
     }
 
     @Test()
-    fun `Test advanced parameters`() {
```

Comment:
Why this test has been removed?

Reply:
Because advanced unpack parameters feature is not supported, and I didn't see any reason left them